File: src/main/typescript/wcardinal/controller/popup-factory.ts
/*
 * Copyright (C) 2019 Toshiba Corporation
 * SPDX-License-Identifier: Apache-2.0
 */

import { Factory } from "./factory";
import { Popup } from "./popup";

/**
 * A popup factory that creates/destroys components dynamically.
 */
export interface PopupFactory extends Factory<Popup> {}

File: src/main/java/org/wcardinal/controller/PopupFactory.java
/*
 * Copyright (C) 2019 Toshiba Corporation
 * SPDX-License-Identifier: Apache-2.0
 */

package org.wcardinal.controller;

/**
 * A factory that creates/destroys the specified class {@code T} as a popup dynamically.
 *
 * @param <T> popup class to be created/destroyed dynamically
 */
public interface PopupFactory<T> extends Factory<T> {

}
